rest/route/github.go

```diff
@@ -269,29 +269,17 @@ func (gh *githubHookApi) Run(ctx context.Context) gimlet.Responder {
 		if event.GetAction() == githubActionRerequested {
 			return gh.handleCheckRunRerequested(ctx, event)
 		}
+
+	case *github.CheckSuiteEvent:
+		if event.GetAction() == githubActionRerequested {
+			return gh.handleCheckSuiteRerequested(ctx, event)
+		}
 	}
 
 	return gimlet.NewJSONResponse(struct{}{})
 }
 
-// handleCheckRunRerequested restarts the task associated with the check run that was rerequested to be re-run and
-// updates the check run to indicate that the task has been restarted.
-func (gh *githubHookApi) handleCheckRunRerequested(ctx context.Context, event *github.CheckRunEvent) gimlet.Responder {
-	owner := event.Repo.Owner.GetLogin()
-	repo := event.Repo.GetName()
-
-	checkRun := event.GetCheckRun()
-	if checkRun == nil {
-		grip.Error(message.Fields{
-			"source":  "GitHub hook",
-			"msg_id":  gh.msgID,
-			"event":   gh.eventType,
-			"owner":   owner,
-			"repo":    repo,
-			"message": "unable to find check run in event",
-		})
-		return gimlet.NewJSONInternalErrorResponse(errors.New("check run not sent from GitHub event"))
-	}
+func (gh *githubHookApi) rerunCheckRun(ctx context.Context, owner, repo string, uid int, checkRun *github.CheckRun) error {
 	checkRunTask := checkRun.GetExternalID()
```

Comment:
Can we please rename this to taskIdFromCheckrun so that it's clear that this is an ID?

Reply:
done